fish_prompt.fish

```diff
@@ -592,46 +592,11 @@ function __bobthefish_prompt_docker -S -d 'Display Docker machine name'
     echo -ns $DOCKER_MACHINE_NAME ' '
 end
 
-function __bobthefish_k8s_context -S -d 'Get the current k8s context'
-    set -l config_paths "$HOME/.kube/config"
-    [ -n "$KUBECONFIG" ]
-    and set config_paths (string split ':' "$KUBECONFIG") $config_paths
-
-    for file in $config_paths
-        [ -f "$file" ]
-        or continue
-
-        while read -l key val
-            if [ "$key" = 'current-context:' ]
-                set -l context (string trim -c '"\' ' -- $val)
-                [ -z "$context" ]
-                and return 1
-
-                echo $context
-                return
-            end
-        end <$file
-    end
-
-    return 1
-end
-
-function __bobthefish_k8s_namespace -S -d 'Get the current k8s namespace'
-    kubectl config view --minify --output "jsonpath={..namespace}"
-end
-
 function __bobthefish_prompt_k8s_context -S -d 'Show current Kubernetes context'
     [ "$theme_display_k8s_context" = 'yes' ]
     or return
 
-    set -l context (__bobthefish_k8s_context)
-    or return
-
-    set -l namespace (__bobthefish_k8s_namespace)
-
-    set -l segment $k8s_glyph " " $context
-    [ -n "$namespace" ]
-    and set segment $segment ":" $namespace
+    set -l segment $k8s_glyph " " (kubectl config view --output "jsonpath={.current-context}:{..namespace}")
```

Comment:
maybe use string to remove it? piping to …

```
string trim --right --chars=:
```

… should do it.

Reply:
It does. Thanks!